Documentation/troubleshooting.md

```diff
@@ -1,4 +1,84 @@
+
+```
+-v value
+    	log level for V logs
+-vmodule value
+    	comma-separated list of pattern=N settings for file-filtered logging
+-log_backtrace_at value
+    	when logging hits line file:N, emit a stack trace
+```
+
+When running under systemd (e.g. on CoreOS container linux) the logs can be viewed with `journalctl -u flanneld`
+
+When flannel is running as a pod on Kubernetes, the logs can be viewed with `kubectl logs --namespace kube-system <POD_ID> -c kube-flannel`. You can find the pod IDs with `kubectl get po --namespace kube-system -l app=flannel`
+
+## Interface selection and the public IP.
+Most backends require that each node has a unique "public IP" address. This address is chosen when flannel starts, and if the address changes then flannel needs to be restarted. Since leases are tied to the public address
+
+The interface chosen and the public IP in use is logged out during startup, e.g.
+```
+I0629 14:28:35.866793    5522 main.go:386] Determining IP address of default interface
+I0629 14:28:35.866987    5522 main.go:399] Using interface with name enp62s0u1u2 and address 172.24.17.174
+I0629 14:28:35.867000    5522 main.go:412] Using 10.10.10.10 as external address
+```
+
+### Vagrant
+Running flannel inside vagrant can cause problems because by default all VMs will have two interfaces. The first is for external traffic that gets NATed. All hosts get the same IP address assigned to to this interface `10.0.2.15`. This causes problems because by default, flannel will select the first interface on a host which leads to all hosts thinking they have the same public IP address. The fix for this is to pass the `--iface eth1` flag to flannel so that the second interface is chosen.
+
+## Permissions
+Depending on the backend being used, flannel may need to run with super user permissions. Examples include creating vxlan devices or programming routes. If you see errors for things like
+ * `Error adding route...`
+ * `Add L2 failed`
+ * `Failed to set up IP Masquerade`
+ * `Error registering network: operation not permitted`
+
+...then confirm that the user running `flannel` has the right permissions (or trying running with `sudo`)
+
+## Performance
+
+### Control plane
+flannel is known to scale to a very large number of hosts. If you see a delay in being able to contact the pods on a newly created host then that can indicate control plane problems. flannel doesn't need much CPU or RAM but the first thing to check would be that it has adaquate resources available. flannel is also reliant on the performance of the datastore, either etcd or the Kubernetes API server. Check that they are performing well.
```

Comment:
cap Flannel at the start of sentences.

... A delay in contacting pods in a newly created host may indicate control plane problems.  ...